apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -68,34 +68,45 @@ require('/modules/publisher.js').exec(function (ctx) {
             var newAsset;
             if (uriMatcher.match(CREATE_URL)) {//to create asset
                 try {
-                    newAsset = assetApi.create(options, req, res, session);
-                    //If the asset is successfully created then return the asset
-                    if (newAsset) {
-                        response = responseProcessor.buildSuccessResponse(res,
-                                constants.STATUS_CODES.CREATED, newAsset)
+                    var contentTypeheader = req.getHeader('Content-Type');
+                    var n = contentTypeheader.indexOf("multipart/form-data");
+                    if (contentTypeheader === 'application/json' || contentTypeheader === 'application/x-www-form-urlencoded' || n >= 0) {
+                        newAsset = assetApi.create(options, req, res, session);
+                        var requestURI = req.getRequestURI();
+                        //If the asset is successfully created then return the asset
+                        if (newAsset) {
+                            response = responseProcessor.buildSuccessResponse(res, constants.STATUS_CODES.CREATED, newAsset);
+                            response.addHeader('Location', requestURI + "/" + newAsset.id + "?type=" + newAsset.type);
+                        } else {
```

Comment:
Try to get rid of nested if-else blocks. It makes code unreadable. Moving codes into separate functions, returning as early as possible etc. might help.